### ISSUE TITLE
Issues around <ssl> and <create_date_time> columns

### ISSUE THREAD

**Issue by prosinger** [turbot/steampipe-plugin-uptimerobot#11]
**Describe the bug**
When I do 

> select create_date_time,ssl from uptimerobot_monitor where type = 1;

I get

+----------------------+--------+
| create_date_time     | ssl    |
+----------------------+--------+
| 1970-01-18T04:55:16Z | <null> |
| 1970-01-19T22:28:28Z | <null> |
| 1970-01-20T05:56:19Z | <null> |

It doesn't seem to read the creation date and if SSL check is active for the monitor. This happnes of course also without filtering out the create_date_time and ssl columns

**Steampipe version (`steampipe -v`)**
Example: v0.3.0

**Plugin version (`steampipe plugin list`)**
Example: v0.5.0

**To reproduce**
select create_date_time,ssl from uptimerobot_monitor where type = 1;

**Expected behavior**
I expected to get creation date/time (time seems to be right) and if the SSL check on the monitor is enabled or not (0/1 for example)

**Additional context**

**Comment by cbruno10** [comment]
Hey @prosinger , thanks for opening this issue!

I was able to reproduce the issue with the `create_date_time` column, and pushed a fixed to the `main` branch (this has not been released yet in a new version).

For SSL, I was unable to reproduce since SSL appears to be a paid feature (or at least I misunderstood where the `ssl` column data should come from), but I've created a branch called `add-raw-data-monitor` that adds a new column called `raw` to the `uptimerobot_monitor` table that should help us debug.

If you get a chance, can you please:
- Clone this repo and switch to the `add-raw-data-monitor` branch
- Run `make` to locally build the plugin
- Then run `select raw from uptimerobot_monitor` and share the results here (with any sensitive info removed)

This column will contain all of the response data we're getting back from the API, so it can help us determine if we're looking in the right field when populating the `ssl` column.

FYI, this branch also contains the `create_date_time` column fix, so please let me know if you still don't see the correct date time showing up.

Thanks!

**Comment by prosinger** [comment]
Ok, give me some time for this please

**Comment by prosinger** [comment]
I tried to compile it but I'm having dependency issues all the time (right now I'm missing redis-dev). I'll need some more time for this, over the weekend or after

**Comment by cbruno10** [comment]
@prosinger What OS are you working on? If you'd like to share the error(s) you're receiving, I'd be happy to try and help debug them. Thanks!

**Comment by prosinger** [comment]
> @prosinger What OS are you working on? If you'd like to share the error(s) you're receiving, I'd be happy to try and help debug them. Thanks!

Ubuntu 20 LTS. I'll follow up on this soon. Sorry for the delay

**Comment by prosinger** [comment]
I managed to compile it on another PC. Apparently the droplet I was tryin on lacks RAM.
So, now I get:
Error: 'api_key' must be set in the connection configuration. Edit your connection configuration file and then restart Steampipe (SQLSTATE HV000)

although the connection is there, and it's read-only

**Comment by prosinger** [comment]
...although this sounds more like it's not being able to find the connection config file, while it's actually there

**Comment by prosinger** [comment]
can you check which config file is it actually expecting (where, name)? It's uptimerobot.spc in the config dir

**Comment by prosinger** [comment]
is one entry enough? (I also have the problem here that it lists only 50 checks)

| {"alert_contacts":null,"create_datetime":1636104766,"friendly_name":"":"","http_username":"","id":789645145,"interval":60,"is_group_main":0,"keyword_case_type":null,"keyword_type":null,"keyword_value":"","logs":null,"monitor_group":0,"port":"","ssl":null,"status":2,"sub_type":"","timeout":30,"type":1,"url":"https://akamai-rap1.somedonain.com/image31.jpg"}

**Comment by cbruno10** [comment]
Thanks @prosinger , so looking at the response data, I can see `ssl` is `null`. Is SSL enabled for that monitor? If so, can you please try running the cURL command (replacing `enterYourAPIKeyHere`):
```
curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -H "Cache-Control: no-cache" -d 'api_key=enterYourAPIKeyHere&format=json&logs=1' "https://api.uptimerobot.com/v2/getMonitors"
```

And see if you get back the correct SSL information for each monitor? If you can find one with SSL disabled and one with it enabled, that will be helpful.

This will help determine if it's the SDK or the API returning `null` for the SSL property.

Thanks!

**Comment by prosinger** [comment]
@cbruno10  yeah Monitor SSL is enabled. This is what I actually notices - you get SSL <null> for all rows in the table

and woah, is there a way to isolate the ecact monitor with the curl? I get  s loy of non-wrapped data when I do your curl format

**Comment by cbruno10** [comment]
@prosinger Yes, you can, the full API documentation is available at https://uptimerobot.com/api/. For the `getMonitors` method, you can specify a specific one like `monitors=15830-32696-83920`.

**Comment by prosinger** [comment]
Here it is:
{"stat":"ok","pagination":{"offset":0,"limit":50,"total":1},"monitors":[{"id":789645145,"friendly_name":"Akamai latency endpoint","url":"https://akamai-rap1.somedomain.com/image31.jpg","type":1,"sub_type":"","keyword_type":null,"keyword_case_type":null,"keyword_value":"","port":"","interval":60,"timeout":30,"status":2,"create_datetime":1636104766,"logs":[{"id":2321806411,"type":2,"datetime":1636562549,"duration":31522323,"reason":{"code":"200","detail":"OK"}},{"id":2320797641,"type":1,"datetime":1636530598,"duration":31951,"reason":{"code":"300001","detail":"The SSL certificate is expired."}},{"id":2307464418,"type":2,"datetime":1636101202,"duration":429396,"reason":{"code":"200","detail":"OK"}},{"id":2307463957,"type":98,"datetime":1636101166,"duration":36,"reason":{"code":"98","detail":"Monitor started"}}],"lastLogTypeBeforeStartDate":{}}]}

**Comment by prosinger** [comment]
ssl | optional (defines if SSL certificate info for each monitor will be returned)
-- | --


so when I add sss=1 into parameters, I get this:

{"stat":"ok","pagination":{"offset":0,"limit":50,"total":1},"monitors":[{"id":789645145,"friendly_name":"Akamai latency endpoint","url":"https://akamai-rap1.somedomain.com/image31.jpg","type":1,"sub_type":"","keyword_type":null,"keyword_case_type":null,"keyword_value":"","port":"","interval":60,"timeout":30,"status":2,"create_datetime":1636104766,"ssl":{"brand":"Let's Encrypt","product":"R3","expires":1674722671,"last_check":"2022-11-10","ignore_errors":0,"disable_notifications":1},"logs":[{"id":2321806411,"type":2,"datetime":1636562549,"duration":31522799,"reason":{"code":"200","detail":"OK"}},{"id":2320797641,"type":1,"datetime":1636530598,"duration":31951,"reason":{"code":"300001","detail":"The SSL certificate is expired."}},{"id":2307464418,"type":2,"datetime":1636101202,"duration":429396,"reason":{"code":"200","detail":"OK"}},{"id":2307463957,"type":98,"datetime":1636101166,"duration":36,"reason":{"code":"98","detail":"Monitor started"}}],"lastLogTypeBeforeStartDate":{}}]}

**Comment by cbruno10** [comment]
Thanks for helping with the debugging @prosinger ! I've released a fix for both of these columns in v0.0.3, if you're still seeing issues, please re-open this issue and let us know.
